core/src/main/java/org/bitcoinj/core/Transaction.java

```diff
@@ -90,9 +92,9 @@ public class Transaction extends ChildMessage {
      * into the past.
      */
     public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME = (tx1, tx2) -> {
-        final long time1 = tx1.getUpdateTime().getTime();
-        final long time2 = tx2.getUpdateTime().getTime();
-        final int updateTimeComparison = -(Long.compare(time1, time2));
+        Instant t1 = tx1.getUpdateTimeInstant().orElse(Instant.EPOCH);
+        Instant t2 = tx2.getUpdateTimeInstant().orElse(Instant.EPOCH);
+        final int updateTimeComparison = -(t1.compareTo(t2));
         //If time1==time2, compare by tx hash to make comparator consistent with equals
         return updateTimeComparison != 0 ? updateTimeComparison : tx1.getTxId().compareTo(tx2.getTxId());
     };
```

Comment:
This can probably be simplified (later) with `Comparators.Comparing`

Reply:
I actually tried that, but so far failed:

```
    public static final Comparator<Transaction> SORT_TX_BY_UPDATE_TIME__ = Comparator
            .comparing(tx -> tx.getUpdateTimeInstant().orElse(Instant.EPOCH), Comparator.reverseOrder())
            .thenComparing(tx -> tx.getTxid());
```

As soon as I add the ".thenComparing", it doesn't recognize tx as type Transaction and I get "Cannot resolve method 'getUpdateTimeInstant' in 'Object'" on the two getters.